### PR TITLE
Developer documentation for InferBound pass.

### PR DESCRIPTION
**Pull request by jdavies-huawei**
An earlier draft of this documentation was discussed here https://discuss.tvm.ai/t/discuss-contributing-new-docs-for-inferbound/2151

@tqchen @jroesch please review, or please suggest reviewers.

### CODE REVIEW

**Comment by tqchen** [comment]
We cannot check-in binary images into the tvm repo, please put that in a separate repo https://github.com/tvmai/tvmai.github.io

**Comment by jdavies-huawei** [comment]
No problem. I created a pull request for the images: https://github.com/tvmai/tvmai.github.io/pull/39

I'll need to update the URLs in the docs, once the images are merged.

**Comment by jdavies-huawei** [comment]
Images have been moved under images/docs/inferbound.

**Comment by jdavies-huawei** [comment]
I updated the URLs to their final versions. So from my perspective this is ready to merge now.

File: docs/dev/codebase_walkthrough.rst [hunk 1]
```diff
@@ -241,3 +241,759 @@ The ``PackedFunc``'s overloaded ``operator()`` will be called, which in turn cal
    };
 
 This concludes an overview of how TVM compiles and executes a function. Although we did not detail TOPI or Relay, at the end all neural network operators go through the same compilation process as above. You are encouraged to dive into the details of the rest of the codebase.
+
+*******************************************
```
[the remark this answers is not in the document]
Comment (jdavies-huawei) [reply docs/dev/codebase_walkthrough.rst]: Ok, I moved this InferBound documentation to its own file. I also added a link from the codebase walkthrough, where it talks about bound inference.

File: docs/dev/codebase_walkthrough.rst [hunk 2]
```diff
@@ -241,3 +241,759 @@ The ``PackedFunc``'s overloaded ``operator()`` will be called, which in turn cal
...
+   }
+
+InferBound with compute_at
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Now that the concept of an attach path has been introduced, we return to how InferBound differs if the schedule contains compute_at. The only difference is in InferRootBound, :ref:`Phase1` and :ref:`Phase2`.
+
+In InferRootBound, the goal is to determine Ranges for the root_iter_vars of a particular stage, C. Phases 1-2 of InferRootBound assign IntSets to the leaf IterVars of C's consumers, and then propagate those IntSets up to the consumers' root_iter_vars.
+
+If there are no attachments, the Ranges already computed for the consumer's variables define how much of C is needed by the consumer. However, if the stage is actually inside the scope of one of the consumer's variables j, then only a single point from within the Range of j is needed at a time.
```
[the remark this answers is not in the document]
Comment (jdavies-huawei) [reply docs/dev/codebase_walkthrough.rst]: I removed all uses of "from within" from the document.

**Comment by vinx13** [comment]
@jdavies-huawei image files are still not removed from this pr?

**Comment by tqchen** [comment]
I removed the commit from the master. please send another PR excluding images

**Pull request merged**